### PR TITLE
refactor: Prepare compile-time check of bilingual format strings

### PR DESCRIPTION
**Pull request by maflcko**
The changes are required for https://github.com/bitcoin/bitcoin/pull/31061, however they also make sense on their own. For example, they are fixing up an `inline namespace`, which lead to compile errors otherwise (can be tested by observing the compile error after reverting the changes to `src/util/strencodings.h`). Also, a unit test comment is fixed.

### CODE REVIEW

**Review by ryanofsky — APPROVED** [review]
Code review ACK fa3e074304780549b1e7972217930e34fa55f59a. Nice changes! These should allow related PRs to be simpler.

**Review by l0rinc — APPROVED** [review]
ACK fa3e074304780549b1e7972217930e34fa55f59a

Lots of versions for this change, loosely coupled, but I understand they will simplify follow-ups.
I have validated manually that:
* `run-lint-format-strings.py` still needs the exception 👍
* all code moves are same as before 👍 
* tidy warns for `feebumper.cpp`, `salvage.cpp` and `wallet.cpp` 👎 (couldn't reproduce any of them, but manually checked it and at least it shouldn't be worse than before)

File: src/wallet/feebumper.cpp [hunk 1]
```diff
@@ -24,24 +24,24 @@ namespace wallet {
 static feebumper::Result PreconditionChecks(const CWallet& wallet, const CWalletTx& wtx, bool require_mine, std::vector<bilingual_str>& errors) EXCLUSIVE_LOCKS_REQUIRED(wallet.cs_wallet)
 {
     if (wallet.HasWalletSpend(wtx.tx)) {
-        errors.push_back(Untranslated("Transaction has descendants in the wallet"));
+        errors.emplace_back(Untranslated("Transaction has descendants in the wallet"));
```
[the remark this answers is not in the document]
Comment (l0rinc) [reply src/wallet/feebumper.cpp:27]: Can you reproduce the warning in this PR? If not, what's the reason for the warning in other PRs (and not here). If you can, what was I doing wrong?

**Review by hodlinator — APPROVED** [review]
cr-ACK fa3e074304780549b1e7972217930e34fa55f59a

#### Before
```C++
strprintf(_("Copyright (C) %i-%i").translated, 2009, COPYRIGHT_YEAR)
```
1. Create a `bilingual_str` of `"Copyright (C) %i-%i"`.
	1. Calls `G_TRANSLATION_FUN("Copyright (C) %i-%i")`, getting something like `"Szerzői jog (c) %i-%i"`, and storing it in `.translated`.
2. Perform regular `strprintf()` (`tfm::format(std::string...)`) on the returned `.translated` and substitute in the values, yielding `"Szerzői jog (c) 2009-2024"`.

#### After
```C++
strprintf(_("Copyright (C) %i-%i"), 2009, COPYRIGHT_YEAR).translated
```
1. Create a `bilingual_str` of `"Copyright (C) %i-%i"`.
	1. Calls `G_TRANSLATION_FUN("Copyright (C) %i-%i")`, getting something like `"Szerzői jog (c) %i-%i"`, and storing it in `.translated`.
2. Perform special `strprintf()` (`tfm::format(bilingual_str...)`).
	1. `tfm::format` call for the `.original` string.
		1. Checks each `args` type, and for `bilingual_str` it fetches the `.original` value for that arg.
	2. `tfm::format` for the `.translated` string.
		1. Checks each `args` type, and for `bilingual_str` it fetches the `.translated` value for that arg.
3. Fetch the `.translated` half of the `strprintf`-result containing `"Szerzői jog (c) 2009-2024"` (throwing away the work from 2.i, `"Copyright (C) 2009-2024"`).

End result is that we do slightly more work at runtime. Translation-strings themselves should not be affected.

**This makes me disagree with the PR summary** that the `.translated` changes in faff8403f0aac3b5ec26d3c7fc98240f879f9906 "make sense on their own" without #31061. But I'm reasonably confident that PR will be merged after having reviewed it as well.

**Pull request merged**